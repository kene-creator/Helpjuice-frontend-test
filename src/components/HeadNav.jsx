import icons from '../assets/icons.svg';
import styles from './HeadNav.module.css';

const HeadNav = () => {
  return (
    <div className={`flex_between ${styles.nav}`}>
      <div className="flex_center">
        <div>
          <svg className={styles.arrow}>
            <use xlinkHref={`${icons}#icon-angle-double-right`} />
          </svg>
        </div>
        <div className="flex_center">
          <svg className={styles.book_open}>
            <use xlinkHref={`${icons}#icon-book-open`} />
          </svg>
          <div className="flex">
            <a
              href="#"
              className="text-[#a4aab4] ml-2 mr-2 border-b border-[#a4aab4] text-[0.7rem]"
            >
              Main
            </a>
            <span className="text-[#a4aab4] text-[0.7rem]">&#47;</span>
            <p className="text-[#a4aab4] mx-2 text-[0.7rem]">Getting Started</p>
            <span className="text-[#a4aab4] text-[0.7rem]">&#47;</span>
            <p className="text-[#a4aab4] mx-2 text-[0.7rem]">
              Front end development test
            </p>
          </div>
        </div>
      </div>
      <div className="flex justify-center items-center">
        <div className="flex mr-4">
          <svg className={styles.book_open}>
            <use xlinkHref={`${icons}#icon-lock`} />
          </svg>
          <p className="text-[#a4aab4] mx-2 text-[0.7rem]">Editing</p>
        </div>
        <span className="text-[#a4aab4] text-[1rem]">|</span>
        <div className="flex justify-center items-center">
          <p className="text-[0.9rem] text-[#3b6aac] mx-2 font-semibold">
            Publish Space
          </p>
          <svg className="rotate-[-90deg] w-[1rem] h-[1rem] fill-[#3b6aac]">
            <use xlinkHref={`${icons}#icon-chevron-left11`} />
          </svg>
        </div>
      </div>
    </div>
  );
};

export default HeadNav;
