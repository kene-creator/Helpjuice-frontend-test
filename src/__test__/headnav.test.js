import React from 'react';
import { render } from '@testing-library/react';
import HeadNav from '../components/HeadNav';

describe('HeadNav component', () => {
  it('renders correctly', () => {
    const wrapper = render(<HeadNav />);
    expect(wrapper).toMatchSnapshot();
  });
});
