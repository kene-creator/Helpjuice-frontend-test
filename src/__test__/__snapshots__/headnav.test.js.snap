// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadNav component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex_between nav"
      >
        <div
          class="flex_center"
        >
          <div>
            <svg
              class="arrow"
            >
              <use
                xlink:href="icons.svg#icon-angle-double-right"
              />
            </svg>
          </div>
          <div
            class="flex_center"
          >
            <svg
              class="book_open"
            >
              <use
                xlink:href="icons.svg#icon-book-open"
              />
            </svg>
            <div
              class="flex"
            >
              <a
                class="text-[#a4aab4] ml-2 mr-2 border-b border-[#a4aab4] text-[0.7rem]"
                href="#"
              >
                Main
              </a>
              <span
                class="text-[#a4aab4] text-[0.7rem]"
              >
                /
              </span>
              <p
                class="text-[#a4aab4] mx-2 text-[0.7rem]"
              >
                Getting Started
              </p>
              <span
                class="text-[#a4aab4] text-[0.7rem]"
              >
                /
              </span>
              <p
                class="text-[#a4aab4] mx-2 text-[0.7rem]"
              >
                Front end development test
              </p>
            </div>
          </div>
        </div>
        <div
          class="flex justify-center items-center"
        >
          <div
            class="flex mr-4"
          >
            <svg
              class="book_open"
            >
              <use
                xlink:href="icons.svg#icon-lock"
              />
            </svg>
            <p
              class="text-[#a4aab4] mx-2 text-[0.7rem]"
            >
              Editing
            </p>
          </div>
          <span
            class="text-[#a4aab4] text-[1rem]"
          >
            |
          </span>
          <div
            class="flex justify-center items-center"
          >
            <p
              class="text-[0.9rem] text-[#3b6aac] mx-2 font-semibold"
            >
              Publish Space
            </p>
            <svg
              class="rotate-[-90deg] w-[1rem] h-[1rem] fill-[#3b6aac]"
            >
              <use
                xlink:href="icons.svg#icon-chevron-left11"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex_between nav"
    >
      <div
        class="flex_center"
      >
        <div>
          <svg
            class="arrow"
          >
            <use
              xlink:href="icons.svg#icon-angle-double-right"
            />
          </svg>
        </div>
        <div
          class="flex_center"
        >
          <svg
            class="book_open"
          >
            <use
              xlink:href="icons.svg#icon-book-open"
            />
          </svg>
          <div
            class="flex"
          >
            <a
              class="text-[#a4aab4] ml-2 mr-2 border-b border-[#a4aab4] text-[0.7rem]"
              href="#"
            >
              Main
            </a>
            <span
              class="text-[#a4aab4] text-[0.7rem]"
            >
              /
            </span>
            <p
              class="text-[#a4aab4] mx-2 text-[0.7rem]"
            >
              Getting Started
            </p>
            <span
              class="text-[#a4aab4] text-[0.7rem]"
            >
              /
            </span>
            <p
              class="text-[#a4aab4] mx-2 text-[0.7rem]"
            >
              Front end development test
            </p>
          </div>
        </div>
      </div>
      <div
        class="flex justify-center items-center"
      >
        <div
          class="flex mr-4"
        >
          <svg
            class="book_open"
          >
            <use
              xlink:href="icons.svg#icon-lock"
            />
          </svg>
          <p
            class="text-[#a4aab4] mx-2 text-[0.7rem]"
          >
            Editing
          </p>
        </div>
        <span
          class="text-[#a4aab4] text-[1rem]"
        >
          |
        </span>
        <div
          class="flex justify-center items-center"
        >
          <p
            class="text-[0.9rem] text-[#3b6aac] mx-2 font-semibold"
          >
            Publish Space
          </p>
          <svg
            class="rotate-[-90deg] w-[1rem] h-[1rem] fill-[#3b6aac]"
          >
            <use
              xlink:href="icons.svg#icon-chevron-left11"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
